adapters/tlsio_openssl.c

```diff
@@ -953,7 +953,7 @@ static int add_certificate_to_store(TLS_IO_INSTANCE* tls_io_instance, const char
         }
         else
         {
-#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && (OPENSSL_VERSION_NUMBER < 0x20000000L)
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && (OPENSSL_VERSION_NUMBER < 0x20000000L) || defined(LIBRESSL_VERSION_NUMBER)
```

Comment:
The change is needed because `OPENSSL_VERSION_NUMBER` is hard-coded to `0x20000000L` in libressl resulting in  `OPENSSL_VERSION_NUMBER` being outside of the accepted range:

https://github.com/libressl-portable/openbsd/blob/7dc88a0dcf7277253dc4e60a13a38cda659f8a81/src/lib/libcrypto/opensslv.h#L11